jtwig-core/src/main/java/org/jtwig/loader/Loader.java

```diff
@@ -0,0 +1,81 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jtwig.loader;
+
+import java.io.InputStream;
+import org.jtwig.exception.ResourceException;
+
+/**
+ * A loader implementation provides a means by which to retrieve a template from
+ * a given datastore, such as a file.
+ */
+public abstract class Loader {
```

Comment:
It most certainly could be. This is just leftover from my experimentations. Will clean up tomorrow.

The relative path is used during the resolution process from within an already parsed template. It is relative to the theoretical root of the twig views directory. It simplifies cross-loader resolution. The canonical path is mostly there for reference purposes, and is also supposed to be helpful where error messaging is required.